dwa_local_planner/src/dwa_planner_ros.cpp

```diff
@@ -184,6 +184,24 @@ namespace dwa_local_planner {
       return false;
     }
 
+    // change limits if dist_tolerance or angle_tolerance are set, and different from the current values
+    if ((dist_tolerance > 0 && dist_tolerance != planner_util_.getCurrentLimits().xy_goal_tolerance) ||
+        (angle_tolerance > 0 && angle_tolerance != planner_util_.getCurrentLimits().yaw_goal_tolerance)) {
+      std::lock_guard<std::mutex> lock(config_mtx_);
+      auto limits = planner_util_.getCurrentLimits();
+      if (dist_tolerance > 0) {
+        ROS_INFO_STREAM_COND(dist_tolerance != limits.xy_goal_tolerance,
+            "updating xy_goal_tolerance to tolerance from action: " << dist_tolerance);
+        limits.xy_goal_tolerance = dist_tolerance;
```

Comment:
Q: What about `inner_xy_goal_tolerance`? Do we change only the outer one? Do we set them both to the same value? I think either is fine depending on what behavior we're looking for when using the action tolerance

Reply:
Disclaimer: the title says `quick and dirty`

So one thing is that unlike the `xy_goal_tolerance` and the `xy_yaw_tolerance`, the `inner_xy_goal_tolerance` won't be reset to the initial value when GGNAS resets the configuration, so I didn't change the `inner_xy_goal_tolerance` value since from looking at the code, having `inner_xy_goal_tolerance` larger than `xy_goal_tolerance` shouldn't cause any issues.

Thinking about it, maybe it would be cleaner to store the configuration set by dynamic reconfigure as the default, and then when the goal is reached, reset to default? Then we could e.g. set the `inner_xy_goal_tolerance` to `min(default_config_.inner_xy_goal_tolerance, dist_tolerance)` ?

Reply:
hmm I'm not sure. I don't like setting it as default because then you lose that option. If someone uses rqt and clicks reset to default, it won't do what's intended. I think it's fine to ignore the inner goal tolerance since it shouldn't cause any issues

Reply:
> Thinking about it, maybe it would be cleaner to store the configuration set by dynamic reconfigure as the default, and then when the goal is reached, reset to default? Then we could e.g. set the inner_xy_goal_tolerance to min(default_config_.inner_xy_goal_tolerance, dist_tolerance) ?

So with this PR we are letting the configured limits permanently changed?
If so, probably makes sense to restore it somehow

Reply:
added restoring the dynamic reconfigure ones in the `isGoalReached()` check if the tolerances in the arguments are 0

Reply:
but leaving inner goal tolerance as is.